src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPriceEstimateTxInsertionManager.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections.Generic;
+using System.Linq;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+
+namespace Nethermind.JsonRpc.Modules.Eth
+{
+    public class GasPriceEstimateTxInsertionManager : ITxInsertionManager
+    {
+        private readonly IGasPriceOracle _gasPriceOracle;
+        private readonly UInt256? _ignoreUnder;
+        private readonly ISpecProvider _specProvider;
+        public GasPriceEstimateTxInsertionManager(IGasPriceOracle gasPriceOracle, UInt256? ignoreUnder, 
+            ISpecProvider specProvider)
+        {
+            _gasPriceOracle = gasPriceOracle;
+            _ignoreUnder = ignoreUnder;
+            _specProvider = specProvider;
+        }
+
+        public int AddValidTxFromBlockAndReturnCount(Block block)
+        {
+            if (block.Transactions.Length > 0)
+            {
+                Transaction[] transactionsInBlock = block.Transactions;
+                int countTxAdded = AddTxAndReturnCountAdded(transactionsInBlock, block);
+
+                if (countTxAdded == 0)
+                {
+                    GetTxGasPriceList().Add((UInt256) _gasPriceOracle.FallbackGasPrice!);
+                    countTxAdded++;
+                }
+
+                return countTxAdded;
+            }
+            else
+            {
+                GetTxGasPriceList().Add((UInt256) _gasPriceOracle.FallbackGasPrice!);
```

Comment:
Cosmetic: I suggest doing:
var txGasPriceList = GetTxGasPriceList();
then you can use txGasPriceList instead of method.